cargo install zebrad fails due to vergen/vergen-lib version conflict

## Summary

`cargo install zebrad` fails to compile with a trait bound error caused by incompatible versions of `vergen-lib` in the dependency tree.

## Error

```
error[E0277]: the trait bound `vergen::feature::cargo::Cargo: vergen_lib::entries::Add` is not satisfied
   --> zebrad-4.0.0/build.rs:21:13
    |
 21 |             &CargoBuilder::all_cargo().expect("all_cargo() should build successfully"),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `vergen_lib::entries::Add` is not implemented for `vergen::feature::cargo::Cargo`
    |
    = note: perhaps two different versions of crate `vergen_lib` are being used?
```

## Root Cause

The workspace specifies `vergen-git2 = { version = "1.0", ... }` which resolves to v1.0.7.

- `vergen-git2` v1.0.7 depends on `vergen` with a semver-compatible range and `vergen-lib` v0.1.6
- When `cargo install` resolves dependencies (without a lock file), it picks the latest compatible `vergen` version: **v9.1.0**
- `vergen` v9.1.0 depends on `vergen-lib` **v9.1.0**

This results in two incompatible versions of `vergen-lib` (v0.1.6 and v9.1.0) in the dependency tree. The `Add` trait from v0.1.6 is incompatible with types from v9.1.0.

**Note:** Building from a cloned repository works fine because `Cargo.lock` pins `vergen` to v9.0.6.

## Suggested Fix

Add an explicit version constraint for `vergen` in the workspace `Cargo.toml`:

```toml
vergen = { version = "9.0", default-features = false }
```

Or pin `vergen-git2` to an exact version:

```toml
vergen-git2 = { version = "=1.0.7", default-features = false }
```

## Environment

- `cargo install zebrad` (version 4.0.0 from crates.io)
- Rust stable

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -153,7 +153,7 @@ opentelemetry = "0.28"
 opentelemetry_sdk = "0.28"
 opentelemetry-otlp = "0.28"
 uint = "0.10"
-vergen-git2 = { version = "1.0", default-features = false }
+vergen-git2 = { version = "9", default-features = false }
 x25519-dalek = "2.0.1"
 zcash_note_encryption = "0.4.1"
 zcash_script = "0.4.2"

